llama_index/node_parser/file/markdown.py

```diff
@@ -0,0 +1,146 @@
+"""Markdown node parser."""
+from typing import List, Optional, Sequence
+
+from llama_index.bridge.pydantic import Field
+from llama_index.callbacks.base import CallbackManager
+from llama_index.callbacks.schema import CBEventType, EventPayload
+from llama_index.node_parser.interface import NodeParser
+from llama_index.schema import BaseNode, TextNode, MetadataMode
+from llama_index.utils import get_tqdm_iterable
+from llama_index.node_parser.node_utils import build_nodes_from_splits
+import re
+
+
+class MarkdownNodeParser(NodeParser):
+    """Markdown node parser.
+
+    Splits a document into Nodes using custom Markdown splitting logic.
+
+    Args:
+        include_metadata (bool): whether to include metadata in nodes
+        include_prev_next_rel (bool): whether to include prev/next relationships
+
+    """
+
+    include_metadata: bool = Field(
+        default=True, description="Whether or not to consider metadata when splitting."
+    )
+    include_prev_next_rel: bool = Field(
+        default=True, description="Include prev/next node relationships."
+    )
+    callback_manager: CallbackManager = Field(
+        default_factory=CallbackManager, exclude=True
+    )
+
+    @classmethod
+    def from_defaults(
+        cls,
+        include_metadata: bool = True,
+        include_prev_next_rel: bool = True,
+        callback_manager: Optional[CallbackManager] = None,
+    ) -> "MarkdownNodeParser":
+        callback_manager = callback_manager or CallbackManager([])
+
+        return cls(
+            include_metadata=include_metadata,
+            include_prev_next_rel=include_prev_next_rel,
+            callback_manager=callback_manager,
+        )
+
+    @classmethod
+    def class_name(cls) -> str:
+        """Get class name."""
+        return "MarkdownNodeParser"
+
+    def get_nodes_from_documents(
```

Comment:
Hmm, feels weird to have `get_nodes_from_documents()` and `get_nodes_from_nodes()` -- but I understand why it's like that lol